plugin/src/create-schema-customization.ts

```diff
@@ -0,0 +1,225 @@
+import { CreateSchemaCustomizationArgs } from "gatsby";
+
+export function createSchemaCustomization(
+  { actions, schema }: CreateSchemaCustomizationArgs,
+  pluginOptions: ShopifyPluginOptions
+) {
+  const includeCollections =
+    pluginOptions.shopifyConnections &&
+    pluginOptions.shopifyConnections.includes("collections");
+
+  // const includeOrders =
+  //   pluginOptions.shopifyConnections &&
+  //   pluginOptions.shopifyConnections.includes("orders");
+
+  // const productImageFields: Fields = {
+  //   product: {
+  //     type: "ShopifyProduct",
+  //
+  //   }
+  //   //product: `ShopifyProduct @link(from: "productId", by: "id")`,
+  // };
+
+  // if (pluginOptions.downloadImages) {
+  //   productImageFields.localFile = `File @link`;
+  // }
+
+  const typeDefs = [
+    schema.buildObjectType({
+      name: "ShopifyProduct",
+      fields: {
+        // collections: includeCollections ? {
+        //   type: "[ShopifyCollection]",
+        //   extensions: {
+        //     directives: [{
+        //       name: "link",
+        //       args: {
+        //         from: "id",
+        //         by: "productIds"
+        //       }
+        //     }]
+        //   }
+        // } : undefined,
+        variants: {
+          type: "[ShopifyProductVariant]",
+          extensions: {
+            directives: [
+              {
+                name: "@link",
+                args: {
+                  from: "id",
+                  by: "productId",
+                },
+              },
+            ],
+          },
```

Comment:
```suggestion
          extensions: {
            link: {
              from: "id",
              by: "productId",
            },
          },
```
Try this (and adjust all other usages). Directives are basically extensions.

Reply:
That certainly is a lot less typing! Thank you!! 🙇🏼